File: Java/ETMSJava/src/main/java/com/example/demo/entities/Employee.java
package com.example.demo.entities;


import jakarta.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


import java.util.List;
import java.util.Objects;

@Entity

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "employee")

public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(nullable = false)
    private String firstName;

    @Column(nullable = false)
    private String lastName;

    @Column(nullable = false)
    private String emailId;

    @Column(nullable = false)
    private int contactNo;

    @Column
    private String address;

    @Column(nullable = false)
    private int loginId;

    @Column(nullable = false)
    private int desigId;

    @ManyToOne
    @JoinColumn(name = "desigId", insertable = false, updatable = false)
    private Designation desig;

    @OneToOne
    @JoinColumn(name = "loginId", insertable = false, updatable = false,unique = true)
    private Login login;

    @OneToMany(mappedBy = "assignedToNavigation", cascade = CascadeType.ALL)
    private List<Project> projectAssignedToNavigations;

    @OneToMany(mappedBy = "createdByNavigation", cascade = CascadeType.ALL)
    private List<Project> projectCreatedByNavigations;

    @OneToMany(mappedBy = "raisedByNavigation", cascade = CascadeType.ALL)
    private List<Query> queries;

    @OneToMany(mappedBy = "solvedByNavigation", cascade = CascadeType.ALL)
    private List<Solution> solutions;

    @OneToMany(mappedBy = "assignedToNavigation", cascade = CascadeType.ALL)
    private List<Task> tasks;

    @OneToOne(mappedBy = "employee", cascade = CascadeType.ALL)
    private TeamMember teamMember;


   
    // hashCode and equals methods
    @Override
    public int hashCode() {
        return Objects.hash(id, firstName, lastName, emailId, contactNo, address, loginId, desigId);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Employee employee = (Employee) o;
        return id == employee.id &&
               contactNo == employee.contactNo &&
               loginId == employee.loginId &&
               desigId == employee.desigId &&
               Objects.equals(firstName, employee.firstName) &&
               Objects.equals(lastName, employee.lastName) &&
               Objects.equals(emailId, employee.emailId) &&
               Objects.equals(address, employee.address);
    }
}

File: DotNet/EtmsSytem/EtmsSytem/Models/Employee.cs
using System;
using System.Collections.Generic;

namespace EtmsSytem.Models;

public partial class Employee
{
    public int Id { get; set; }

    public int LoginId { get; set; }

    public string FirstName { get; set; } = null!;

    public string LastName { get; set; } = null!;

    public int ContactNo { get; set; }

    public string EmailId { get; set; } = null!;

    public string Address { get; set; } = null!;

    public int DesigId { get; set; }

    public virtual Designation Desig { get; set; } = null!;

    public virtual Login Login { get; set; } = null!;

    public virtual ICollection<Project> ProjectAssignedToNavigations { get; set; } = new List<Project>();

    public virtual ICollection<Project> ProjectCreatedByNavigations { get; set; } = new List<Project>();

    public virtual ICollection<Query> Queries { get; set; } = new List<Query>();

    public virtual ICollection<Solution> Solutions { get; set; } = new List<Solution>();

    public virtual ICollection<Task> Tasks { get; set; } = new List<Task>();

    public virtual ICollection<TeamMember> TeamMembers { get; set; } = new List<TeamMember>();
}
